proxy/grpcproxy/kv.go

```diff
@@ -51,19 +51,30 @@ func (p *kvProxy) Range(ctx context.Context, r *pb.RangeRequest) (*pb.RangeRespo
 		return nil, err
 	}
 
-	p.cache.Add(r, (*pb.RangeResponse)(resp.Get()))
+	// cache linearizable as serializable
+	r.Serializable = true
+	gresp := (*pb.RangeResponse)(resp.Get())
+	p.cache.Add(r, gresp)
 
-	return (*pb.RangeResponse)(resp.Get()), nil
+	return gresp, nil
 }
 
 func (p *kvProxy) Put(ctx context.Context, r *pb.PutRequest) (*pb.PutResponse, error) {
 	resp, err := p.kv.Do(ctx, PutRequestToOp(r))
-	return (*pb.PutResponse)(resp.Put()), err
+	if err == nil {
```

Comment:
this may need to be unconditional since the put could go through but lose the response